src/test/java/ua/com/juja/microservices/teams/dao/TeamRepositoryTest.java

```diff
@@ -0,0 +1,151 @@
+package ua.com.juja.microservices.teams.dao;
+
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
+import ua.com.juja.microservices.integration.BaseIntegrationTest;
+import ua.com.juja.microservices.teams.entity.Team;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.inject.Inject;
+
+import java.time.*;
+import java.util.*;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Ivan Shapovalov
+ * @author Andrii Sidun
+ */
+@RunWith(SpringRunner.class)
+public class TeamRepositoryTest extends BaseIntegrationTest {
+
+    @Inject
+    private TeamRepository teamRepository;
+
+    @Test
+    public void test_saveTeamExecutedCorrectly(){
+        final String userInOneTeam = "user-in-one-team";
+        final String userInSeveralTeams = "user-in-several-teams";
+        final Team expected = new Team(new HashSet<>(Arrays.asList(userInOneTeam, "user1", "user2", userInSeveralTeams)));
+
+        Team actual = teamRepository.saveTeam(expected);
+
+        assertNotNull(actual);
+        assertThat(actual.getMembers(), is(expected.getMembers()));
+    }
+
+    @Test
+    @UsingDataSet(locations = "/datasets/getAndDeactivateDataSet.json")
+    public void test_getUserTeamsUserInOneTeamExecutedCorrectly() {
+        Date actualDate = Date.from(Instant.now());
+        final String userInOneTeam = "user-in-one-team";
+        final String userInSeveralTeams = "user-in-several-teams";
+        final Team expected = new Team(new HashSet<>(Arrays.asList(userInOneTeam, "user1", "user2", userInSeveralTeams)));
+
+        List<Team> actual = teamRepository.getUserActiveTeams(userInOneTeam,actualDate);
+
+        assertEquals(actual.size(),1);
+        assertThat(actual.get(0).getMembers(), is(expected.getMembers()));
+    }
+
+    @Test
+    @UsingDataSet(locations = "/datasets/getAndDeactivateDataSet.json",loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
+    public void test_getUserTeamsIfUserInSeveralTeamsExecutedCorrectly() {
+        Date actualDate = Date.from(Instant.now());
+        final String userInOneTeam = "user-in-one-team";
+        final String userInSeveralTeams = "user-in-several-teams";
+        final Team team1 = new Team(new LinkedHashSet<>(Arrays.asList(userInOneTeam, "user1", "user2", userInSeveralTeams)));
+        final Team team2 = new Team(new LinkedHashSet<>(Arrays.asList(userInSeveralTeams, "user3", "user4", "user5")));
+        final List<Team> expected = new ArrayList<>();
+        expected.add(team1);
+        expected.add(team2);
+
+        List<Team> actual = teamRepository.getUserActiveTeams(userInSeveralTeams, actualDate);
+
+        assertEquals(actual.size(), expected.size());
+        for (int i = 0; i < actual.size(); i++) {
```

Comment:
If the Team entity has equals and hashcode overrided methods, you can check these entities with `assertEquals(expected, actual)` only